causallearn/utils/KCI/KCI.py

```diff
@@ -426,11 +438,11 @@ def kernel_matrix(self, data_x, data_y, data_z):
                 # construct Gaussian kernels according to learned hyperparameters
                 Kzy = gpy.kernel_.k1(data_z, data_z)
                 self.epsilon_y = np.exp(gpy.kernel_.theta[-1])
-        elif self.kernelY == 'Polynomial':
+        elif self.kernelZ == 'Polynomial':
```

Comment:
this seems an important bug to me --- the old code doesn't really work I guess?

Reply:
Yes it's a typo bug. `KernelY` should be `KernelZ`.

However, in practical usage, people usually choose the same kernel for both X, Y, and for Z. And thus this bug happens to be avoided so far.

Reply:
Cool, good to know! 

If we have tests earlier, we would detect this bug. ;)

@aoqiz A good case why good tests are needed. :)

Reply:
Yes, thanks.